src/encoded/audit/experiment_series.py

```diff
@@ -0,0 +1,167 @@
+from snovault import (
+    AuditFailure,
+    audit_checker,
+)
+from .formatter import (
+    audit_link,
+    path_to_text,
+)
+
+
+def generate_formatted_list_of_experiments(list_of_experiments):
+	if len(list_of_experiments) < 3:
+		formatted_list = ' and '.join([audit_link(path_to_text(item),item) for item in list_of_experiments])
+	else:
+		formatted_list = '{}, and {}'.format(
+			', '.join([audit_link(path_to_text(item),item) for item in list_of_experiments[0:-1]]),
+			audit_link(path_to_text(list_of_experiments[-1]),list_of_experiments[-1])
+			)
+	return formatted_list
+
+
+@audit_checker('ExperimentSeries', frame=['related_datasets',
+                                          'related_datasets.replicates',
+                                          'related_datasets.replicates.library',
+                                          'related_datasets.replicates.library.biosample',
+                                          'related_datasets.replicates.library.biosample.treatments',
+                                          ])
+def audit_mismatched_properties(value, system):
+	excluded_statuses = ['deleted', 'replaced', 'revoked']
+	assays = dict()
+	biosample_types = dict()
+	targets = {'no target': []}
+	donors = dict()
+	treatments = {'no treatment': []}
+	genetic_modifications = {'no genetic modification': []}
+
+	filtered_datasets = [dataset for dataset in value['related_datasets'] if dataset['status'] not in excluded_statuses]
+
+	for dataset in filtered_datasets:
+		if dataset['assay_title'] in assays:
+			assays[dataset['assay_title']].append(dataset['@id'])
+		else:
+			assays[dataset['assay_title']] = [dataset['@id']]
+
+		if dataset['biosample_ontology'] in biosample_types:
+			biosample_types[dataset['biosample_ontology']].append(dataset['@id'])
+		else:
+			biosample_types[dataset['biosample_ontology']] = [dataset['@id']]
+
+		if 'target' not in dataset:
+			targets['no target'].append(dataset['@id'])
+		else:
+			if dataset['target'] in targets:
+				targets[dataset['target']].append(dataset['@id'])
+			else:
+				targets[dataset['target']] = [dataset['@id']]	
+
+		if 'replicates' in dataset:
+			for replicate in dataset['replicates']:
+				if replicate['status'] not in ['deleted'] and \
+				'library' in replicate and 'biosample' in replicate['library']:
+					biosample = replicate['library']['biosample']
```

Comment:
I'm not sure but
1. Should the status check against `excluded_statuses`? I mean replicate can be "revoked" (cannot be "replaced" though)?
2. Should we check status on library and biosample as well?

Reply:
Spoke with Idan and he agrees status should be checked for replicate/library/biosample. I will update this.